### PR TITLE
Deduplication

### PR DESCRIPTION
**Pull request by recursivepaws**
This branch solves deduplication. Prior to this code implementation, duplicate files were being detected, but ignored in the packing step. As a result, filesystems with duplicate files were not able to be processed. This code addressed these issues by handling duplicate files, while only packing the original file. Test cases have been written to ensure this functionality.

### CODE REVIEW

**Comment by recursivepaws** [comment]
the test failing here is `test_deduplication_size` which is designed to ensure packed filesystems with 50% duplicates are 50% of the footprint of filesystems with only unique files, *when they are packed*. This relies on an exact assertion that, while functional on my local machine, may not be exactly accurate in all environments. A better test condition would be to ensure that the packed dup filesystem is at least a certain factor smaller than its packed unique counterpart, rather than an exact factor. This can be updated in future versions.

File: dataprep-pipelines/src/plan_copy.rs [hunk 1]
```diff
@@ -28,73 +28,102 @@ use crate::utils::hasher;
 pub async fn plan_copy(
     origin_data: SpiderMetadata,
     to_root: PathBuf,
-    seen_hashes: Arc<RwLock<HashMap<String, Rc<SpiderMetadata>>>>,
+    // This data structure is a little uglier than i would like it to be
+    seen_hashes: Arc<RwLock<HashMap<String, DuplicationMetadata>>>,
```
Comment (amiller68) [inline dataprep-pipelines/src/plan_copy.rs:32]: Was this reference Counter important here?

Comment (amiller68) [reply dataprep-pipelines/src/plan_copy.rs:32]: Tests pass so, i'm sure this is fine

**Review by amiller68 — DISMISSED** [review]
This looks good!

**Review by amiller68 — APPROVED** [review]
Looks good!

**Pull request merged**